packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -259,6 +259,17 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
         }
     }
     
+    // Initial Save fails for QueryPredicate.all, so we must pass nil instead
```

Comment:
The root cause is initial save cannot be called with a query predicate.  In Amplify-iOS if you try to make an initial save with any queryPredicate it will fail. 

For Amplify-Android initial save will fail for any queryPredicate except for .all

Reply:
Thanks!

If we want to merge this iOS specific trick in, could we add a comment to link to the related amplify-ios? So we can do a follow up to clean up when the underlying iOS issue gets fixed.

Reply:
Sure I can add a small comment so we remember to fix this.

Reply:
Is there an issue on the ios repo? If so, it would be nice to include it in the comment. That way in 3 years from now when no one has context on this anymore, anyone can just open a link, see the issue was resolved (hopefully), and know that it is safe to update the code 🙂

Reply:
Created an issue and linked to it